### PR TITLE
First pass at making on-prem product-generic

### PR DESCRIPTION
**Pull request by predominant**
Signed-off-by: Graham Weldon <graham@grahamweldon.com>

This needs discussion, and a lot of review.

Calling all Jan Michael Vincents! @chefsalim 

tenor-207945461

### CODE REVIEW

**Comment by chefsalim** [comment]
@predominant Thanks for the PR - will get to this shortly

**Comment by chefsalim** [comment]
Will add more feedback but on first run through initial feedback - this seems like quite a bit of change to absorb all at once. Changes to the migrate scripts especially are not easy to test, so unless you want to invest effort into testing those scenarios and signing off that they still work, we probably should roll those back. Another thing to consider is that we will likely move to a different installer in the not too distant future, so the utility of these changes may not be worthwhile.

File: install.sh [hunk 1]
```diff
@@ -3,30 +3,21 @@
...
-
-if [ "$license" == "accept" ] || [ "$license" == "accept-no-persist" ]; then
-  read -r -p "Continue with installation? [y/N] " response
-else
-  curl https://raw.githubusercontent.com/habitat-sh/on-prem-builder/master/LICENSE-NOTICE
-  echo
-
-  read -r -p "Do you accept the terms of this license? Answering yes will proceed with the installation. [y/N] " response
-fi
+source license.sh
```
Comment (chefsalim) [inline install.sh]: I don't see a license.sh in the PR..

File: scripts/provision.sh [hunk 2]
```diff
@@ -11,12 +11,9 @@ else
   exit 1
 fi
 
-# Defaults
-BLDR_ORIGIN=${BLDR_ORIGIN:="habitat"}
```
Comment (chefsalim) [inline scripts/provision.sh]: Does this get set somewhere else now?

File: scripts/provision.sh [hunk 3]
```diff
@@ -68,7 +65,7 @@ EOT
     FEATURES_ENABLED=""
     ARTIFACTORY_API_URL="http://localhost:8081"
     ARTIFACTORY_API_KEY="none"
-    ARTIFACTORY_REPO="habitat-builder-artifact-store"
+    ARTIFACTORY_REPO="builder-artifact-store"
```
Comment (chefsalim) [inline scripts/provision.sh]: This could make migration difficult - I'd suggest not changing the name

File: scripts/sup.service.sh [hunk 4]
```diff
@@ -14,17 +14,17 @@ if [ ! -z "$NO_PROXY" ]; then
 Environment=\"NO_PROXY=${NO_PROXY}\""
 fi
 if [ -z "$SSL_CERT_FILE" ]; then
-  SSL_CERT_FILE="$(hab pkg path core/cacerts)/ssl/cert.pem"
+  SSL_CERT_FILE="$(${HAB_CMD} pkg path core/cacerts)/ssl/cert.pem"
 fi
 
-cat <<EOT > /etc/systemd/system/hab-sup.service
+cat <<EOT > /etc/systemd/system/sup.service
```
Comment (chefsalim) [inline scripts/sup.service.sh]: May be better to keep some prefix descriptor in addition to just "sup"

File: install.sh [hunk 5]
```diff
@@ -3,30 +3,21 @@
...
 
+cat NOTICE
+read -r -p "Continue with installation? [y/N] " response
 if [[ "$response" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
-    pushd scripts > /dev/null
-    sudo ./install-hab.sh
-    sudo ./hab-sup.service.sh
-    sudo ./provision.sh
-    popd > /dev/null
+  source settings.env
```
Comment (chefsalim) [inline install.sh]: I don't see a settings.env in the PR...

**Comment by predominant** [comment]
This was premature. Will look at this again in the coming weeks. Thanks for the honest review @chefsalim - Much appreciated!

**Pull request closed without merging**